cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -132,6 +126,18 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
    * @throws IOException if there is a problem in making an HTTP request to check if the namespace exists.
    */
   public DatasetMeta get(Id.DatasetInstance instance, List<? extends Id> owners) throws Exception {
+    try {
+      return metaCache.get(instance);
+    } catch (ExecutionException e) {
+      Throwables.propagateIfPossible(e.getCause(), Exception.class);
```

Comment:
E.g. In this sample code:

```java
  private static final Logger LOG = LoggerFactory.getLogger(SimpleTest.class);

  public void get() throws Throwable {
    LoadingCache<String, String> cache = CacheBuilder.newBuilder().build(new CacheLoader<String, String>() {
      @Override
      public String load(String key) throws Exception {
        throw new Exception("Key is not valid");
      }
    });

    try {
      cache.get("Hello");
    } catch (ExecutionException e) {
      Throwables.propagateIfPossible(e.getCause(), Exception.class);
      throw Throwables.propagate(e.getCause());
    }
  }

  @Test
  public void test() {
    try {
      get();
    } catch (Throwable t) {
      LOG.error("Error: {}", t.getClass(), t);
    }
  }
```

The log looks like this:

```
2016-06-08 23:15:05,509 - ERROR [main:c.c.c.SimpleTest@57] - Error: class java.lang.Exception
java.lang.Exception: Key is not valid
	at co.cask.cdap.SimpleTest$1.load(SimpleTest.java:40) ~[test-classes/:na]
	at co.cask.cdap.SimpleTest$1.load(SimpleTest.java:37) ~[test-classes/:na]
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3589) ~[guava-13.0.1.jar:na]
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2374) ~[guava-13.0.1.jar:na]
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2337) ~[guava-13.0.1.jar:na]
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2252) ~[guava-13.0.1.jar:na]
	at com.google.common.cache.LocalCache.get(LocalCache.java:3990) ~[guava-13.0.1.jar:na]
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3994) ~[guava-13.0.1.jar:na]
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4878) ~[guava-13.0.1.jar:na]
	at co.cask.cdap.SimpleTest.get(SimpleTest.java:45) ~[test-classes/:na]
	at co.cask.cdap.SimpleTest.test(SimpleTest.java:55) ~[test-classes/:na]
```

and it doesn't tell it is actually from the `get()` method.